File: Snooper-Spring-No-Op/src/main/java/org/springframework/http/client/SnooperInterceptor.java
package org.springframework.http.client;

import org.springframework.http.HttpRequest;

import java.io.IOException;

public class SnooperInterceptor implements ClientHttpRequestInterceptor {
  @Override
  public ClientHttpResponse intercept(HttpRequest httpRequest, byte[] bytes, ClientHttpRequestExecution execution) throws IOException {
    return execution.execute(httpRequest, bytes);
  }
}

File: Snooper-Spring-No-Op/src/main/java/org/springframework/http/client/SnooperInterceptor.kt
package org.springframework.http.client

import org.springframework.http.HttpRequest

import java.io.IOException

class SnooperInterceptor : ClientHttpRequestInterceptor {
  @Throws(IOException::class)
  override fun intercept(
    httpRequest: HttpRequest,
    bytes: ByteArray,
    execution: ClientHttpRequestExecution
  ): ClientHttpResponse {
    return execution.execute(httpRequest, bytes)
  }
}
